source/app/requesthandlers/scheduler_setup_handler.py

```diff
@@ -331,7 +409,31 @@ def _create_request(self):
         return self._update_settings() and self.set_lambda_logs_retention_period()
 
     def _update_request(self):
-        return self._update_settings() and self.set_lambda_logs_retention_period()
+        try:
+            org_id = list(self.remote_account_ids)[0]
+        except Exception as error:
+            self._logger.info(f"org id is not valid or empty {error}")
+            org_id = ""
+        try:
+            prev_org_id = list(self.old_remote_account_ids)[0]
+        except Exception as error:
+            self._logger.info(
+                f"org id from old custom resource request parameters is not valid or empty {error}"
+            )
+            prev_org_id = ""
+        if (
+            self.is_valid_org_id(org_id)
+            and self.is_valid_org_id(prev_org_id)
+            and org_id == prev_org_id
+        ):
+            config = configuration.get_scheduler_configuration(self._logger)
+            prev_remote_account_id = config.remote_account_ids
+        else:
+            prev_remote_account_id = {}
+        return (
+            self._update_settings(prev_remote_account_id)
+            and self.set_lambda_logs_retention_period()
+        )
```

Comment:
The control flow here is quite hard to follow. It would be nice to validate things earlier in the process so we don't have to be this paranoid later